src/messageschannel.cpp

```diff
@@ -97,7 +102,38 @@ MatrixMessagesChannelPtr MatrixMessagesChannel::create(MatrixConnection *connect
 
 QString MatrixMessagesChannel::sendMessageCallback(const Tp::MessagePartList &messageParts, uint flags, Tp::DBusError *error)
 {
-    return QString();
+    if (!m_room) {
+        // TODO: error?
+        qDebug() << Q_FUNC_INFO;
+        return QString();
+    }
+    QString content;
+    for (auto &part : messageParts) {
+        // TODO handle other parts?
+        if(part.count(QStringLiteral("content-type")) &&
+                part.value(QStringLiteral("content-type")).variant().toString() ==
+                    QLatin1String("text/plain") &&
+                part.count(QStringLiteral("content"))
+        ) {
+            content = part.value(QStringLiteral("content")).variant().toString();
+            break;
+        }
+    }
+    auto txnId = m_room->postPlainText(content);
+    auto& pending = m_room->pendingEvents();
+    if (pending.empty()) {
+        // TODO: error?
+        qDebug() << Q_FUNC_INFO;
+        return QString();
+    }
+    auto it = std::find_if(pending.begin(), pending.end(),
+                           [&txnId] (const auto& e) {
+        return e->transactionId() == txnId;
+    });
+    // "Message sending": it != pending.end();
+    // TODO: Wait when it actually gets sent; check that it obtained an id
```

Comment:
This TODO is actually not related to sendMessage method.
Instead we need to change message delivery status to [Accepted](https://telepathy.freedesktop.org/spec/Channel_Interface_Messages.html#Enum:Delivery_Status). It seems similar to Telegram message send. At first, the client assign random 64 bit message identifier and then receive respond from server with the message sequential 32 bit id (that is actually used to reference the message).
See also: https://github.com/TelepathyIM/telepathy-morse/blob/next/textchannel.cpp#L414

Reply:
Thank you! Tried to implement status support, looks like telepathy logger is showing delivered messages now.